src/coreComponents/mesh/generators/WellGeneratorBase.cpp

```diff
@@ -528,54 +531,53 @@ void WellGeneratorBase::debugWellGeometry() const
     return;
   }
 
-  std::cout << std::endl;
-  std::cout << "++++++++++++++++++++++++++" << std::endl;
-  std::cout << "WellGeneratorBase = " << getName() << std::endl;
-  std::cout << "MPI rank = " << MpiWrapper::commRank( MPI_COMM_GEOSX ) << std::endl << std::endl;
-  std::cout << "Number of well elements = " << m_numElems << std::endl;
+  //1. formatting data
+  TableLayout tableWellLayout = TableLayout( {
+      TableLayout::ColumnParam{"Element no.", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"CoordX", TableLayout::Alignment::middle},
+      TableLayout::ColumnParam{"CoordY", TableLayout::Alignment::middle},
+      TableLayout::ColumnParam{"CoordZ", TableLayout::Alignment::middle},
+      TableLayout::ColumnParam{"Prev\nElement", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"Next\nElement", TableLayout::Alignment::right},
+    } );
+
+  tableWellLayout.setTitle( "InternalWellGenerator " + getName());
```

Comment:
- you can swap `1. formatting data` and `2. collecting data` as this `1.` step is closer to `3. dumping`
- you may have forgotten to adress [a comment of Thomas](https://github.com/GEOS-DEV/GEOS/pull/2984#discussion_r1508095225). Adding the title to the constructor of `TableLayout` :
  * will allow you to set `tableWellLayout` and `tableLayoutPerfo` as constants,
  * will allow this syntax :
```c++
  TableTextFormatter tablePerfoLog( { {"Perforation no.", "Coordinates", "connected to" },
                                      "Peforation table" } );
```
  (not saying that you must use it here, `TableLayout` variables can help for readability, do as you prefer)